### PR TITLE
made a map of map to be passed to the frontend

### PR DESCRIPTION
**Pull request by ahujatejas06**
Updated the output for proforma questions

### CODE REVIEW

File: application/admin.questions.go [hunk 1]
```diff
@@ -138,15 +138,12 @@ func getAnswersForProforma(ctx *gin.Context, pid uint) map[uint][]ApplicationAns
 	if err != nil {
 		ctx.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 	}
-	var returnedAnswerArray = make(map[uint][]ApplicationAnswerResponse)
+	var returnedAnswerArray = make(map[uint](map[uint]string))
```
Comment (1-Harshit) [inline application/admin.questions.go:141]: Ref 1

**Comment by 1-Harshit** [comment]
```go
func getAnswersForProforma(ctx *gin.Context, pid uint) map[uint](map[uint]string) {
	var questions []ApplicationQuestion

	err := fetchProformaQuestion(ctx, pid, &questions)

	if err != nil {
		ctx.AbortWithStatusJSON(http.StatusExpectationFailed, gin.H{"error": err.Error()})
	}

	var questionID []uint
	for _, ques := range questions {
		questionID = append(questionID, ques.ID)
	}

	var answers []ApplicationQuestionAnswer

	err = fetchAllAnswers(ctx, pid, questionID, &answers)
	if err != nil {
		ctx.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": err.Error()})
	}

	var answers_map = make(map[uint](map[uint]string))

	for _, ans := range answers {
		qid := ans.ApplicationQuestionID
		sid := ans.StudentRecruitmentCycleID

		if answers_map[sid] == nil {
			answers_map[sid] = make(map[uint]string)
		}

		answers_map[sid][qid] = ans.Answer
	}
	return answers_map
}```

Here's the updated code of what you wanted. Carefully go over it, and commit only if you're totally convinced.

File: application/admin.questions.go (hunk 1 above)
Comment (ahujatejas06) [reply application/admin.questions.go:141]: Yes got it... The second map isn't getting initialized before use. Only a pointer to it exists without allocating any memory

File: application/admin.questions.go [hunk 2]
```diff
@@ -117,7 +117,7 @@ type ApplicationAnswerResponse struct {
 	Answer     string `json:"answer"`
```
Comment (1-Harshit) [inline application/admin.questions.go:117]: just remove this we're good to merge

**Review by 1-Harshit — APPROVED** [review]

**Pull request merged**